### ISSUE TITLE
Demons Souls Vulkan renderer freeze

### ISSUE THREAD

**Issue by Krusher97** [RPCS3/rpcs3#3116]
OS: Windows 7 64bits
CPU: i5 6500 @3.2GHz
GPU: Nvidia 750Ti 2GB
RAM: 8GB

The game freezes either in the intro or during the tutorial level when using the vulkan renderer. 
I got to the nexus using the OpenGL one without encountering any freezes. 
![demons settigns](https://user-images.githubusercontent.com/25664221/28707366-b8dd1dc8-7395-11e7-9bb7-10aca701f872.png)

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/1182330/RPCS3.zip)

**Comment by isshininu** [comment]
Same issue.

**Comment by heffokrg** [comment]
I got the same issue. 

AMD r9 390 (17.7.1) 
2500k @ 4.5GHz
8GB Ram
Win 7 x64
Vulkan updated to 1.0.54.0

**Comment by Gncd** [comment]
I have similar issues, I've got the game to run the longest on rpcs3-v0.0.3-2017-07-23-386ff781 build
Heres a my log file from rpcs3-v0.0.3-2017-07-29-0a4776d5 
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/1184720/RPCS3.zip)

**Comment by squerol** [comment]
Same situation here.

RPCS3 v0.0.3-5-0a4776d Alpha
Latest Vulkan Libraries (from latest Nvidia Driver Update)

Windows 7 Ultimate x64 SP1
Intel Core i5 7500 3,4 Ghz Quad-Core
MSI GTX 1060 Gaming X Plus 6GB GDDR5
16 GB DDR4 RAM
Gigabyte GA-B250M-D3H

[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/1184799/RPCS3.zip)

**Comment by isshininu** [comment]
On Windows 10 i didn't encounter freeze on Vulkan.

And as you can see, all bug reports are related to W7 only.

**Comment by ajoneil** [comment]
I have experience the freeze, and I am on Windows 10. I think the lle-gcm merge a couple of hours ago was supposed to fix it though, I haven't given it a test yet.

**Comment by isshininu** [comment]
No, lle-gcm have nothing to fix.

**Comment by Krusher97** [comment]
tested it again with the latest build but still got random freezes.

**Comment by ymgve** [comment]
I got Vulkan freezes too, and it seems to be related to the driver version.

Windows 7 64bit, i5-3550, 16GB RAM, GTX980

Nvidia driver 372.54 - NO freezes, but shadows are bugged
Nvidia driver 382.05 - freezes
Nvidia driver 384.94 - freezes

When it's freezing it's generally seconds into the tutorial cutscene (dude walking in water towards a light) - some times audio continues playing for a while, some times everything stops.

I've also noticed that when it freezes in this way, using the "Stop emulation" button in the GUI doesn't properly stop emulation - the video output window still sticks around.

**Comment by ymgve** [comment]
Here's some more testing I've done with various driver versions. Definitely seems like something changed starting with version 378.49. (This is all tested on RPCS3 build 0.0.3-5588)

    geforce         vulkanrt
    372.54          1.0.11.1        no crashes, broken shadows
    375.63          1.0.26.0        no crashes, broken shadows
    376.09          1.0.26.0        no crashes, broken shadows
    376.09          1.0.26.0        no crashes, broken shadows
    376.33          1.0.26.0        no crashes, broken shadows
    378.49          1.0.37.0        crashes
    382.05          1.0.42.1        crashes
    384.94          1.0.42.1        crashes

**Comment by ap-Di0** [comment]
The build **rpcs3-v0.0.3-2017-07-25-0302e8da_win64** works well for me at good speeds (im about halfway through the game on it), any newer builds will hang the game a few minutes after loading any 3D on Vulkan.

**Comment by ArtyomAN** [comment]
> The build rpcs3-v0.0.3-2017-07-25-0302e8da_win64 works well for me at good speeds

Sorry for silly question... Could you please show me where I can download this version for testing? I can't find 0302e8da here: https://rpcs3.net/compatibility?b.

**Comment by Oceangraves** [comment]
https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.3-5455/artifacts

**Comment by Krusher97** [comment]
Freezes for me even in that build and with the vulkan version 1.0.26(but with driver version 384.94)

**Comment by marii-moe** [comment]
I may be having a similar issue with Atelier Ayesha's menu - and I've bisected the issue to #3083 with Vulkan renderer.

https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.3-5479/artifacts is the last working version for me.
https://ci.appveyor.com/project/rpcs3/rpcs3/build/0.0.3-5487/artifacts is the version where #3083 was committed.

Does this issue reproduce on 5487 but not on 5479?

**Comment by Krusher97** [comment]
Happens for me even in 5479.

**Comment by kd-11** [comment]
This is a windows 7 bug. Some games with vulkan renderers also do not recommend vulkan be used on windows 7 with GPUs with less than 4G of VRAM on nvidia GPUs for this same reason.
The windows 10 crashes do not seem related to the main issue mentioned by the OP

**Comment by paulsapps** [comment]
what is the reason for the hang? when out of vram it hangs on w7 but not on win10?

**Comment by kd-11** [comment]
Its very unlikely that we overcommit, VRAM usage hovers around 500M. I'm not entirely sure why but I remember when doom first started supporting vulkan, windows 7 was not supported for a very similar reason. Something about the internals on win7 doesnt play nice with vulkan. Maybe we're running out of handles? I have no idea.

**Comment by Krusher97** [comment]
Can't just be vulkan. The new Doom worked for me and Dolphin works pretty well with vulkan on Windows 7.

I haven't tried many games but stuff like Dragons Crown and Tokyo Jungle ran for longer without freezing.

**Comment by kd-11** [comment]
Yea, that's usually how these things go. Everything works fine until some condition is met that triggers the bug. If you have access to a debugger you can connect to the emulator and try to figure out where it hangs - the driver probably never returns and we wait infinitely for a timeout.

**Comment by kd-11** [comment]
Also, you can run the game with "debug output" + strict mode enabled. It will run at very low fps but its the only way to figure out if something broke. Without validation traces nvidia will be unlikely to follow up on any complaints.

**Comment by Megamouse** [comment]
Is this still happening? There was no update for a long time and there have been many changes since then

**Comment by isshininu** [comment]
Well, i've enciuntered this several builds ago, but i think this regression already fixed (again), since games no more freezes after start (again).

**Comment by kd-11** [comment]
Reopen if issue persists

**Comment by Krusher97** [comment]
The emulator didn't freeze like it used to(audio playing while the game get stuck) but I got this error message while fighting the vanguard which crashed the emulator.
![rpcs3](https://user-images.githubusercontent.com/25664221/32146929-a137000e-bd04-11e7-8328-26a8479a4e18.png)
On my second run I rushed to him and managed to get to the nexus without encountering any freezes. Seems like the issue is fixed for me. Not sure if its the emulator itself or the latest Nvidia driver update that did it.
